corehq/apps/app_manager/static/app_manager/js/widgets.js

```diff
@@ -56,24 +56,23 @@ hqDefine("app_manager/js/widgets", [
             width: options.width || '200px',
         });
 
-        if ($('#app-profile-id-input').length) {
+        if ($('#build-profile-id-input').length) {
             $select.on('select2:select', function (e) {
                 var buildProfiles = _.map(e.params.data.buildProfiles, function (details, profileId) {
                     return { id: profileId, text: details.name };
                 });
-                $('#app-profile-id-input').select2({
-                    placeholder: gettext("Select profile"),
-                    allowClear: true,
-                    data: buildProfiles,
-                });
+                if (buildProfiles.length) {
+                    $('#build-profile-id-input').append(new Option(gettext('Select Profile'), '', true, true));
```

Comment:
Nitpick: it's be more semantic to pass a `placeholder` option to the select2 call.

Reply:
i remember we talked about that option on a PR earlier and then decided to keep this considering few factors. I am happy to link that if needed. So keeping this.

Reply:
Now that you mention it, that does sound familiar... :)